templates/base_index.html

```diff
@@ -1409,55 +1409,103 @@ <h2>
+          selectedIndex = Math.floor(Math.random() * selectedTakeovers.length);
+          showTakeover(selectedTakeovers, selectedIndex);
+        }
 
-    if (selectedTakeovers) {
-      var selectedIndex = null;
+        // Store the current takeover
+        localStorage.setItem('selected_takeover_index', selectedIndex)
 
-      if (localStorage.getItem('selected_takeover_index') !== null) {
-        // If we previously chose a takeover, increment the number to show the next takeover
-        var nextIndex = parseInt(localStorage.getItem('selected_takeover_index')) + 1
-        selectedIndex = nextIndex < selectedTakeovers.length ? nextIndex : 0;
-      } else {
-        // Otherwise, randomly choose one of the takeovers and store it for next time
-        selectedIndex = Math.floor(Math.random() * selectedTakeovers.length);
+        }
       }
+    };
+
+    xhr.open("GET", "/takeovers.json", true);
+    xhr.send();
 
-      // Store the current takeover
-      localStorage.setItem('selected_takeover_index', selectedIndex)
+  }
 
-      // Get the takeover element
-      var selectedTakeover = selectedTakeovers[selectedIndex]
+  function showTakeover(takeovers, index) {
+    // Default parameter
+    index = typeof index !== 'undefined' ? index : 0;
 
-      if (! document.body.contains(selectedTakeover)) {
-        // If it's not in the current document (as in, it's in a template) import it
-        // Note: This is for all browser *except* IE, which doesn't support <template>s
-        selectedTakeover = document.importNode(selectedTakeover, true);
-      }
+    // Get HTML elements
+    var takeover = document.getElementById("takeover");
+    var title = document.getElementById("takeover-title");
+    var subtitle = document.getElementById("takeover-subtitle");
+    var image = document.getElementById("takeover-image");
+    var primaryUrl = document.getElementById("takeover-primary-url");
+    var secondaryUrl = document.getElementById("takeover-secondary-url");
+    var get_ctas = document.getElementById("takeover-ctas").getElementsByTagName("p")[0];
+    var takeoverAnimation = document.getElementsByClassName("p-takeover-animation")
+
+    // Set values to homepage takeover
+    var selectedTakeover = takeovers[index];
+
+    takeover.className = "";
+    takeover.removeAttribute("style");
+
+    // Add takeover classes
+    var classNameString = "js-takeover p-takeover--" + selectedTakeover.class;
+    takeover.className += classNameString;
+
+    // Set language attributes
+    takeover.setAttribute("lang", selectedTakeover.lang);
+    takeover.setAttribute("lang-skip", selectedTakeover.lang_skip);
+
+    // Set takeover content
+    title.textContent = selectedTakeover.title;
+    subtitle.textContent = selectedTakeover.subtitle;
+    image.src = selectedTakeover.image;
+    image.removeAttribute("style");
+    image.style.width = selectedTakeover.image_width;
```

Comment:
I think this might not be overriding the parameter `width=""`, perhaps the default should have `style="width:200px;"` not `width=""`

Reply:
I don't see how this is changing the markup. Is it correct?

Reply:
Done, yes this is correct.